src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -538,24 +538,21 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
           this.feedbackQuestionModels.set(updatedQuestion.feedbackQuestionId, updatedQuestion);
           this.loadResponseStatusForQuestion(this.questionEditFormModels[index]);
 
-          // shift question if needed
+          // reload questions in new order if needed
           if (originalQuestionNumber !== updatedQuestion.questionNumber) {
-            // move question form
-            this.moveQuestionForm(
-                originalQuestionNumber - 1, updatedQuestion.questionNumber - 1);
-            this.normalizeQuestionNumberInQuestionForms();
+            this.reloadQuestionForms();
           }
 
           this.statusMessageService.showSuccessToast('The changes to the question have been updated.');
         }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
   }
 
   /**
-   * Moves question edit form from the original position to the new position.
+   * Reloads feedback questions.
```

Comment:
I think we need to explain the workaround a bit as this violate our design in frontend where we want to minimize the unnecessary communication.